testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveDatabricksHttpMetastore.java

```diff
@@ -0,0 +1,86 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.hive;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.tempto.BeforeMethodWithContext;
+import io.trino.testng.services.Flaky;
+import io.trino.tests.product.deltalake.BaseTestDeltaLakeS3Storage;
+import org.testng.annotations.Test;
+
+import java.util.Locale;
+
+import static io.trino.tempto.assertions.QueryAssert.Row.row;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static io.trino.tests.product.TestGroups.DATABRICKS_UNITY_HTTP_HMS;
+import static io.trino.tests.product.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static io.trino.tests.product.deltalake.util.DeltaLakeTestUtils.DATABRICKS_COMMUNICATION_FAILURE_ISSUE;
+import static io.trino.tests.product.deltalake.util.DeltaLakeTestUtils.DATABRICKS_COMMUNICATION_FAILURE_MATCH;
+import static io.trino.tests.product.deltalake.util.DeltaLakeTestUtils.getRandomPathInExternalLocation;
+import static io.trino.tests.product.utils.QueryExecutors.onDelta;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestHiveDatabricksHttpMetastore
+        extends BaseTestDeltaLakeS3Storage
+{
+    private String unityCatalogCatalogName;
+    private String externalLocationPath;
+
+    @BeforeMethodWithContext
+    public void setUp()
+    {
+        super.setUp();
+        unityCatalogCatalogName = requireNonNull(System.getenv("DATABRICKS_UNITY_CATALOG_NAME"), "Environment variable not set: DATABRICKS_UNITY_CATALOG_NAME");
+        externalLocationPath = requireNonNull(System.getenv("DATABRICKS_UNITY_EXTERNAL_LOCATION"), "Environment variable not set: DATABRICKS_UNITY_EXTERNAL_LOCATION");
+    }
+
+    @Test(groups = {DATABRICKS_UNITY_HTTP_HMS, PROFILE_SPECIFIC_TESTS})
+    @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
+    public void testBasicHiveOperations()
+    {
+        String schemaName = "testBasicHiveOperations_" + randomNameSuffix();
+        try {
+            String tableName = schemaName + "." + "testTable";
+            String tableLocation = getRandomPathInExternalLocation(bucketName, externalLocationPath, tableName);
+            String schemaLocation = getRandomPathInExternalLocation(bucketName, externalLocationPath, schemaName);
+            onDelta().executeQuery(format(
```

Comment:
```
Suppressed: java.lang.Exception: Query: CREATE SCHEMA mycatalog.testBasicHiveOperations_qcxcv0jwan MANAGED LOCATION 's3://trino-ci-test/test_databricks_unity/48568843-c3fe-4618-97ed-c8e9c6dab17a/external/testBasicHiveOperations_qcxcv0jwan/a71ce24c-e2af-49c2-ada9-b7a5ec0a8317'
tests               | 		at io.trino.tempto.query.JdbcQueryExecutor.executeQueryNoParams(JdbcQueryExecutor.java:136)
tests               | 		at io.trino.tempto.query.JdbcQueryExecutor.execute(JdbcQueryExecutor.java:112)
tests               | 		at io.trino.tempto.query.JdbcQueryExecutor.executeQuery(JdbcQueryExecutor.java:84)
tests               | 		at io.trino.tests.product.utils.QueryExecutors$3.lambda$executeQuery$0(QueryExecutors.java:137)
tests               | 		at dev.failsafe.Functions.lambda$toCtxSupplier$11(Functions.java:243)
tests               | 		at dev.failsafe.Functions.lambda$get$0(Functions.java:46)
tests               | 		at dev.failsafe.internal.RetryPolicyExecutor.lambda$apply$0(RetryPolicyExecutor.java:74)
tests               | 		at dev.failsafe.SyncExecutionImpl.executeSync(SyncExecutionImpl.java:187)
tests               | 		at dev.failsafe.FailsafeExecutor.call(FailsafeExecutor.java:376)
tests               | 		at dev.failsafe.FailsafeExecutor.get(FailsafeExecutor.java:112)
tests               | 		at io.trino.tests.product.utils.QueryExecutors$3.executeQuery(QueryExecutors.java:137)
tests               | 		at io.trino.tests.product.hive.TestHiveDatabricksHttpMetastore.testBasicHiveOperations(TestHiveDatabricksHttpMetastore.java:60)
tests               | 		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
tests               | 		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
tests               | 		at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
tests               | 		at java.base/java.lang.reflect.Method.invoke(Method.java:568)
tests               | 		at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
tests               | 		at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
tests               | 		at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
tests               | 		at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
tests               | 		at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
tests               | 		at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
tests               | 		at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
tests               | 		at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
tests               | 		at java.base/java.lang.Thread.run(Thread.java:833)
tests               | Caused by: com.databricks.client.support.exceptions.GeneralException: [Databricks][DatabricksJDBCDriver](500051) ERROR processing query/statement. Error Code: 0, SQL state: org.apache.hive.service.cli.HiveSQLException: Error running query: com.databricks.sql.managedcatalog.UnityCatalogServiceException: [RequestId=d0996c13-43e8-4eb3-832d-a3e417d56c1c ErrorClass=INVALID_PARAMETER_VALUE] CreateSchema storage_root length should be less than 150
```
locations are too long

Reply:
We should fit under this constraint by doing shorter schema names and table names.

```
s3://trino-ci-test/test_databricks_unity/48568843-c3fe-4618-97ed-c8e9c6dab17a/external/test_schema_qcxcv0jwan/test_basic_qcxcv0jwan
```

Reply:
I believe the latest commit does not have this problem anymore.